modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkSystem.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.core.logic.trunk;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.audio.AudioManager;
+import org.terasology.audio.events.PlaySoundEvent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.metadata.EntitySystemLibrary;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.inventory.ItemComponent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.Region3i;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.In;
+import org.terasology.rendering.logic.MeshComponent;
+import org.terasology.utilities.Assets;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.entity.placement.PlaceBlocks;
+import org.terasology.world.block.regions.BlockRegionComponent;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class TrunkSystem extends BaseComponentSystem {
+    private static final Logger logger = LoggerFactory.getLogger(TrunkSystem.class);
+
+    @In
+    private WorldProvider worldProvider;
+    @In
+    private BlockEntityRegistry blockEntityRegistry;
+    @In
+    private EntityManager entityManager;
+    @In
+    private AudioManager audioManager;
+    @In
+    private InventoryManager inventoryManager;
+    @In
+    private EntitySystemLibrary entitySystemLibrary;
+
+    @ReceiveEvent(components = {TrunkComponent.class, ItemComponent.class})
+    public void placeTrunk(ActivateEvent event, EntityRef entity) {
```

Comment:
Could you please split this up into smaller (private) methods to slim this method down a bit?

Reply:
could also be done in a follow-up PR, tbh.